AppCenterData/AppCenterData/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -189,6 +189,106 @@ - (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString
                                           fromDeviceCache:YES];
 }
 
+- (MSPaginatedDocuments *)listWithToken:(MSTokenResult *)token
+                              partition:(NSString *)partition
+                           documentType:(Class)documentType
+                            baseOptions:(MSBaseOptions *_Nullable)baseOptions {
+
+  // Final list of documents.
+  NSMutableArray<MSDocumentWrapper *> *localListItems = [NSMutableArray new];
+
+  // Get effective device time to live.
+  NSInteger deviceTimeToLive = baseOptions ? baseOptions.deviceTimeToLive : kMSDataTimeToLiveDefault;
+
+  // Execute the query.
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:token.partition];
+  NSString *selectionQuery =
+      [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\"", tableName, kMSPartitionColumnName, token.partition];
+  NSArray *listResult = [self.dbStorage executeSelectionQuery:selectionQuery];
+  NSDate *currentDate = [NSDate date];
+
+  // Parse the documents.
+  for (id documentRow in listResult) {
+
+    // If an expired document is found exclude it from the list and delete it from the local store.
+    long expirationTime = [(NSNumber *)(documentRow[self.expirationTimeColumnIndex]) longValue];
+    NSString *documentId = documentRow[self.documentIdColumnIndex];
+    if (expirationTime != kMSDataTimeToLiveInfinite) {
+      NSDate *expirationDate = [NSDate dateWithTimeIntervalSince1970:expirationTime];
+      if (expirationDate && [expirationDate laterDate:currentDate] == currentDate) {
+        NSString *warningMessage =
+            [NSString stringWithFormat:@"Local document for partition '%@' and document ID '%@' expired at %@, discarding it",
+                                       token.partition, documentId, expirationDate];
+        MSLogWarning([MSData logTag], @"%@", warningMessage);
+
+        // Delete the local document when found to be expired
+        [self deleteWithToken:token documentId:documentId];
+      } else if ([kMSPendingOperationDelete isEqualToString:documentRow[self.pendingOperationColumnIndex]]) {
+
+        // Ignore document, if the pending operation is found to be Delete
+        NSString *warningMessage = [NSString
```

Comment:
Why don't you just use `MSLogError` to format instead of introducing a new variable?